modules/deseq2/differential/main.nf

```diff
@@ -0,0 +1,46 @@
+params.deseq_gene_id_col = "gene_id"
```

Comment:
Ahh, thanks- I knew this wasn't quite right, but I was having trouble seeing how it should be done properly given the template situation, and I didn't see any other usages of module.config in the nf-core modules.

Reply:
@SPPearce sorry, just to clarify, by module.config do you mean the pipeline-level [modules.config](https://github.com/nf-core/rnaseq/blob/master/conf/modules.config), or is there something at the module level I can use to set the defaults?

Reply:
The module level yes.
Generally the `ext.args` option is used to access this. If you look in the `tests/modules/` files you can see how it is then used.
The pipelines have a config file in `conf/modules.config` that controls these options for each module.

Reply:
@SPPearce Just to double-check:

**What I have now**

```
params.deseq_gene_id_col = "gene_id"
```

Which I can retrieve from within the template like:

```
opt <- list(
    ...
    'gene_id_col' = '$params.deseq_gene_id_col',
...
)
```

I assume that param value could be overridden in calling workflows. 

**What should be happening**

`task.ext.args` is a single string, right? So I'd have to parse the 'gene_id' value out of a longer string like `--deseq_gene_id_col column_A --foo bar --another-option blah`.

Because I'm using a template rather than an existing CLI tool, I then have to assign that variable back to something that can be seen from the template- which I think probably means `params`. 

That seems like a lot of trouble to avoid using params in the first place :-D

Reply:
When you aren't using a template, you'd just have this sort of set up: 
`deseq_de.r $ext.args`
where the `modules.config` will specify e.g.
`ext.args = '--gene_id_col "gene_id" --sample_id_col = "experiment_accession" --test = "Wald" --fit_type = "parametric"'`
and you'd use an argument parsing function inside the R script. 
So then in a pipeline, the modules.config can be customised (and may well refer to external params for them!). 
I haven't used template personally (I've used the bin folder in our local pipeline to do this), I assume that you can't do  e.g. `template 'my_script.R -a foo -b bar'?

Reply:
No, I don't think they work that way. Even if they did I'd have to compose the CLI for the script, which would require r-optparse, which would make the software dependency handling harder (right now I can just use the bioconductor-deseq2 biocontainer, which is very convenient). Plus I'm not sure what the purpose would be of contriving that CLI to pass parameters that can be read directly by the template.

Essentially I think the  params/ template usage is justified in situations like this where we're wrapping libraries rather than command line tools, which from poking around in this repo seems quite novel for nf-core (though I can see myself using the pattern repeatedly).

The test pipeline that's in place now illustrates the usage, in that parameter settings can be overridden from the calling workflow and read directly by the template. I really don't see the need to stuff those params into an arg string and parse them back out again for the template (though it makes complete sense in the more conventional case where args are passed through to a command line tool).